hal/sparcv8leon3/gaisler/timer.c

```diff
@@ -24,14 +24,20 @@
 #define TIMER_INT_PENDING (1 << 4)
 #define TIMER_CHAIN       (1 << 5)
 
-#define TIMER_DEFAULT_FREQ 1000
-
+/* Timer registers */
 
-/* clang-format off */
+#define GPT_SCALER     0       /* Scaler value register                 : 0x00 */
+#define GPT_SRELOAD    1       /* Scaler reload value register          : 0x04 */
+#define GPT_CONFIG     2       /* Configuration register                : 0x08 */
+#define GPT_LATCHCFG   3       /* Latch configuration register          : 0x0C */
+#define GPT_TCNTVAL(n) (n * 4) /* Timer n counter value reg (n=1,2,...) : 0xn0 */
+#define GPT_TRLDVAL(n) (n * 4) /* Timer n reload value register         : 0xn4 */
```

Comment:
Shouldn't be `((n * 4) + 1)`?

Reply:
In the documentation timers are numbered from 1, so as bad as it is I think we should stay true to it.

Reply:
Ok, but if you want to get 0xn4 from n, then the register index should be 4n + 1, shouldn't it?